src/main/java/org/wildfly/security/permission/PermissionUtil.java

```diff
@@ -435,13 +435,19 @@ public static Permission createPermission(final Class<? extends Permission> perm
             }
         }
         try {
-            if (twoArg != null) {
+            if (twoArg != null && name != null && actions != null) {
```

Comment:
Only the permission implementation class can know what arguments are required or what their constraints are, unfortunately.